api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -0,0 +1,112 @@
+      // it may lead to path confusion vulnerabilities.
+      UNESCAPE_AND_FORWARD = 4;
+    }
+
+    // Should paths be normalized according to RFC 3986 before any processing of
+    // requests by HTTP filters or routing? This affects the upstream *:path* header
+    // as well. For paths that fail this check, Envoy will respond with 400 to
+    // paths that are malformed. The default behavior is to normalize URI path.
+    // This value may be overridden by the runtime variable
+    // :ref:`http_connection_manager.normalize_path<config_http_conn_man_runtime_normalize_path>`.
+    // See `Normalization and Comparison <https://tools.ietf.org/html/rfc3986#section-6>`_
+    // for details of normalization.
+    // Note that Envoy does not perform
+    // `case normalization <https://tools.ietf.org/html/rfc3986#section-6.2.2.1>`_
+    bool skip_path_normalization = 1;
+
+    // Determines if adjacent slashes in the path are merged into one before any processing of
+    // requests by HTTP filters or routing. This affects the upstream *:path* header as well. Setting
+    // this option to true will cause incoming requests with path `//dir///file` to not match against route
+    // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
+    // `HTTP spec <https://tools.ietf.org/html/rfc3986>`_ and is provided for convenience.
+    bool skip_merging_slashes = 2;
+
+    // Action to take when request URL path contains escaped slash sequences (%2F, %2f, %5C and %5c).
+    // The default value can be overridden by the :ref:`http_connection_manager.path_with_escaped_slashes_action<config_http_conn_man_runtime_path_with_escaped_slashes_action>`
+    // runtime variable.
+    // The :ref:`http_connection_manager.path_with_escaped_slashes_action_sampling<config_http_conn_man_runtime_path_with_escaped_slashes_action_enabled>` runtime
+    // variable can be used to apply the action to a portion of all requests.
+    PathWithEscapedSlashesAction path_with_escaped_slashes_action = 3;
+  }
+
+  message Http1ProtocolOptions {
+    // Allows Envoy to process HTTP/1 requests/responses with both `Content-Length` and `Transfer-Encoding`
+    // headers set. By default such messages are rejected, but if option is enabled - Envoy will
+    // remove Content-Length header and process message.
+    // See `RFC7230, sec. 3.3.3 <https://tools.ietf.org/html/rfc7230#section-3.3.3>`_ for details.
+    //
+    // .. attention::
+    //   Enabling this option might lead to request smuggling vulnerability, especially if traffic
+    //   is proxied via multiple layers of proxies.
+    bool allow_chunked_length = 1;
+
+    // Allow extension methods in HTTP/1 requests. By default this option is turned off and only
+    // HTTP methods defined in the `RFC 7231 section 4.1 <https://tools.ietf.org/html/rfc7231#section-4.1>`_
+    // are allowed.
+    bool allow_extension_methods = 2;
```

Comment:
did our http/2 parser restrict?
I'd be inclined to allow most methods by default, as I think HTTP parser was overly restrictive.

Reply:
I did not. I agree. I have restructured it, such that all methods are accepted by default and can be optionally restricted.